multi-agent/adapters/outbound/mongo/blueprint_repository.py

```
@@ -39,19 +48,25 @@ def save(self, identity: Identity, spec: BlueprintDraft,
             "rid_refs": rid_refs,
             "metadata": metadata,
         }
-        self._col.insert_one(doc)
+        try:
+            self._col.insert_one(doc)
+        except DuplicateKeyError:
+            raise BlueprintDuplicateNameError(spec.name)
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)